OpenRA.Mods.Cnc/Traits/PortableChrono.cs

```diff
@@ -102,19 +103,26 @@ public Order IssueOrder(Actor self, IOrderTargeter order, Target target, bool qu
 
 		public void ResolveOrder(Actor self, Order order)
 		{
-			if (order.OrderString == "PortableChronoTeleport" && CanTeleport && order.Target.Type != TargetType.Invalid)
+			if (order.OrderString == "PortableChronoTeleport" && order.Target.Type != TargetType.Invalid)
 			{
 				var maxDistance = Info.HasDistanceLimit ? Info.MaxDistance : (int?)null;
-				self.CancelActivity();
+				if (!order.Queued)
+					self.CancelActivity();
 
 				var cell = self.World.Map.CellContaining(order.Target.CenterPosition);
+				var move = self.Trait<IMove>();
```

Comment:
You can save the trait lookup overhead by caching this from the constructor. Note that you'd normally do this from `INotifyCreated` but the `Requires<IMoveInfo>` guarantees that the trait you want exists when the constructor runs.

Reply:
Fixed.